instrumentation/runtime-metrics/library/README.md

```diff
@@ -0,0 +1,270 @@
+
+All metrics reported by the service are backed by
+[asynchronous instruments](https://opentelemetry.io/docs/reference/specification/metrics/api/#synchronous-and-asynchronous-instruments) which can be a
+[Counter](https://opentelemetry.io/docs/reference/specification/metrics/api/#asynchronous-counter),
+[UpDownCounter](https://opentelemetry.io/docs/reference/specification/metrics/api/#asynchronous-updowncounter), or a
+[Gauge](https://opentelemetry.io/docs/reference/specification/metrics/api/#asynchronous-gauge) (the default).
+
+To figure out what MBeans (or ObjectNames) and their attributes are available for your system, check its documentation, or use a universal MBean browsing tool, such as `jconsole`, available for every JDK version.
+
+### Composite Types
+
+The next example shows how the current heap size can be reported.
+```yaml
+---
+rules:
+  - bean: java.lang:type=Memory
+    mapping:
+      HeapMemoryUsage.used:
+        metric: jvm.heap.used
+        type: updowncounter
+        desc: The current heap size
+        unit: B
+      HeapMemoryUsage.max:
+        metric: jvm.heap.max
+        type: updowncounter
+        desc: The maximum allowed heap size
+        unit: B
+```
+The MBean responsible for memory statistics, identified by ObjectName `java.lang:type=Memory` has an attribute named `HeapMemoryUsage`, which is of a `CompositeType`. This type represents a collection of fields with values (very much like the traditional `struct` data type). To access individual fields of the structure we use a dot which separates the MBean attribute name from the field name. The values are reported in bytes, which here we indicate by `B`. In the above example, the current heap size and the maximum allowed heap size will be reported as two metrics, named `jvm.heap.used`, and `jvm.heap.max`.
+
+### Measurement Attributes
+
+A more advanced example shows how to report similar metrics related to individual memory pools. A JVM can use a number of memory pools, some of them are part of the heap, and some are for JVM internal use. The number and the names of the memory pools depend on the JVM vendor, the Java version, and may even depend on the java command line options. Since the memory pools, in general, are unknown, we will use wildcard character for specifying memory pool name (in other words, we will use what is known as an ObjectName pattern).
```

Comment:
Memory pool instrumentation is already covered by [MemoryPools](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/MemoryPools.java) which follows the  [JVM runtime semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/runtime-environment-metrics.md#jvm-metrics). 

So we wouldn't really want a user to separately instrument their memory pools. But thinking about it, if this evolves to be able to connect to remote JVMs then it would be useful to add JMX based memory pool monitoring. In that case, we'd want to have predefined configuration which matches the semantic conventions.

Reply:
It was never my intention to encourage users to do that. This is just an example for the user manual. Memory pools were selected, because at least some users have familiarity with these concepts, so it is easier to read and understand.

Reply:
I updated the documentation, stressing that the quoted YAML code snippets are just examples.